src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -1033,6 +1033,18 @@ static StageInfo getStageInfo(JsonNode jsonNode) throws SnowflakeSQLException {
       }
     }
 
+    boolean useS3RegionalUrl;
+    if (stageInfo.getStageType() == StageInfo.StageType.S3) {
+      // This node's value is set if PUT is used without Session. (For Snowpipe Streaming, we rely
+      // on a response from a server to have this field set to use S3RegionalURL)
+      JsonNode useS3RegionalURLNode =
+          jsonNode.path("data").path("stageInfo").path("useS3RegionalUrl");
+      if (!useS3RegionalURLNode.isMissingNode()) {
+        useS3RegionalUrl = useS3RegionalURLNode.asBoolean(false);
```

Comment:
How to check this only affect without session? What happen in with session code? It may override this value for with session. Will it cause any regression?

Reply:
Good point, thanks for pointing that out.
I can pass in session object in getStageInfo to distinguish session vs sessionless and only set this when it is sessionless. WDYT?

Reply:
sounds good.

Reply:
Made changes as suggested, PTAL! Thanks